### PR TITLE
Issue-24918: Favorite pages: print the error message from the HTML2Canvas lib to the console

### LINKED ISSUE #24916: Favorite Pages: Unable to create bookmark image for /members page
### Parent Issue

#22343 

### Problem Statement

I am trying to create a bookmark for a page that does not have CMS_ANONYMOUS view on it and we cannot seem to get a thumbnail of it.

Screen Shot 2023-05-11 at 1 45 01 PM

### Steps to Reproduce

To reproduce:
- in edit mode, browse to "members only" page.
- try to make a bookmark for it.
- You will see the preview fail to generate.

### Acceptance Criteria

Be able to create a thumbnail for any page in the system.

### dotCMS Version

23.06

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### LINKED ISSUE #24918: Favorite pages: print the error message from the HTML2Canvas lib to the console
### Parent Issue

#22343 

### Task

Let's add the error message when the HTML2Canvas library fail to the console, let's do a `console.warn` and with the message:

```
We apologize for the inconvenience. The following error occurred while generating a thumbnail for this page:

${ERROR}
```

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

User should see an warning message in the console when the HTML2Canvas failed to generate the thumbnail.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by manuelrojas**
### <samp>🤖 Generated by Copilot at 2550ba3</samp>

Improved the functionality and appearance of the `dot-favorite-page` component and the `dot-html-to-image` component. Fixed an unused import and added error handling in the former, and enhanced iframe loading and rendering in the latter. Adjusted the height and position of the components to avoid extra space and overlapping.

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots

image

image

image

https://github.com/dotCMS/core/assets/3438705/fad952b3-c6fc-4e85-8443-2ca41e82e3aa

https://github.com/dotCMS/core/assets/3438705/7d45d3cc-00b2-4db0-9183-53ced004f616

### CODE REVIEW

**Review by AndreyDotcms — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts [hunk 1]
```diff
@@ -165,6 +164,10 @@ export class DotFavoritePageComponent implements OnInit, OnDestroy {
                 } else {
                     this.form.get('thumbnail').setValue('');
                     this.store.setShowFavoriteEmptySkeleton(true);
+                    console.warn(
+                        'We apologize for the inconvenience. The following error occurred while generating a thumbnail for this page:',
+                        event?.detail?.error
+                    );
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts:170]: Should we make this message `multi-language`? Something like:

```suggestion
                    console.warn(
                        this.dotMessagePipe.transform('favorite.pages.generate.thumbnail.error'),
                        event?.detail?.error
                    );
```

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts:170]: This is just a suggestion. Feel free to ignore it If you don't think it is worthy it.

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts (hunk 1 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/dot-favorite-page.component.ts:170]: test? not sure if we can do a spy on `console.warn`

**Review by alfredo-dotcms — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

File: core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.tsx [hunk 2]
```diff
@@ -34,6 +36,8 @@ export class DotHtmlToImage {
         }).then((canvas) => {
             canvas.toBlob((blob) => {
                 const fileObj = new File([blob], 'image.png');
+                const iframe = parent.document.querySelector('#${this.iframeId}');
```
[the remark this answers is not in the document]
Comment (manuelrojas) [reply core-web/libs/dotcms-webcomponents/src/components/dot-html-to-image/dot-html-to-image.tsx:39]: It's been generated correctly: 

image

**Review by alfredo-dotcms — APPROVED** [review]

**Pull request merged**